logdna/resource_alert_test.go

```diff
@@ -154,19 +154,19 @@ func TestAlert_ErrorsSlackChannel(t *testing.T) {
 func TestAlert_ErrorsWebhookChannel(t *testing.T) {
 	btArgs := map[string]map[string]string{"webhook": cloneDefaults(chnlDefaults["webhook"])}
 	btArgs["webhook"]["bodytemplate"] = `"{\"test\": }"`
-	btCfgE := fmtTestConfigResource("alert", "new", nilLst, alertDefaults, btArgs)
+	btCfgE := fmtTestConfigResource("alert", "new", nilLst, alertDefaults, btArgs, nilLst)
 
 	mdArgs := map[string]map[string]string{"webhook": cloneDefaults(chnlDefaults["webhook"])}
 	mdArgs["webhook"]["method"] = `"false"`
-	mdCfgE := fmtTestConfigResource("alert", "new", nilLst, alertDefaults, mdArgs)
+	mdCfgE := fmtTestConfigResource("alert", "new", nilLst, alertDefaults, mdArgs, nilLst)
```

Comment:
Adding all of these `nilList` because the function signature changed seems really painful and the parameter list is too large at 6 params, IMO.  It's a bigger change, but we should be using a config structure for this.  Then, we could easily default values.

That's the problem with these test functions being abstracted.  It gets really messy trying to make it work for everything.

If you're up to it, make that a new commit.  Create a `testResource{}` structure and add all the things.  `fmtTestConfigRequest` can then be called with that structure and only worry about formatting the properties that are defined.

Fine to create a card to fix later, cause I think we need to get this easier to use before it becomes out of hand.

Reply:
It makes sense. I'll work on this refactoring in a separate story